File: Editorial/150_Evaluate_Reverse_Polish_Notation/Go/2_Evaluate_with_Stack.go
func evalRPN(tokens []string) int {
    operations := map[string]func(int, int) int{
        "+": func(a, b int) int { return a + b },
        "-": func(a, b int) int { return a - b },
        "*": func(a, b int) int { return a * b },
        "/": func(a, b int) int { return a / b },
    }
    stack := []int{}
    for _, token := range tokens {
        if operation, exists := operations[token]; exists {
            num1 := stack[len(stack)-2]
            num2 := stack[len(stack)-1]
            stack = stack[:len(stack)-2]
            result := operation(num1, num2)
            stack = append(stack, result)
        } else {
            num, _ := strconv.Atoi(token)
            stack = append(stack, num)
        }
    }
    return stack[0]
}

File: Editorial/150_Evaluate_Reverse_Polish_Notation/TypeScript/2_Evaluate_with_Stack.ts
function evalRPN(tokens: string[]): number {
    const OPERATORS = {
        "+": (a: number, b: number) => a + b,
        "-": (a: number, b: number) => a - b,
        "/": (a: number, b: number) => Math.trunc(a / b),
        "*": (a: number, b: number) => a * b,
    };
    const stack: number[] = [];
    for (const token of tokens) {
        if (token in OPERATORS) {
            const number2 = stack.pop() || 0;
            const number1 = stack.pop() || 0;
            const operation = OPERATORS[token];
            stack.push(operation(number1, number2));
        } else {
            stack.push(Number(token));
        }
    }
    return stack.pop() || 0;
}
